algorithm.go

```diff
@@ -164,3 +173,20 @@ func (a Algorithm) FromBytes(p []byte) Digest {
 func (a Algorithm) FromString(s string) Digest {
 	return a.FromBytes([]byte(s))
 }
+
+// ValidateEncoded validates the encoded portion string
+func (a Algorithm) ValidateEncoded(encoded string) error {
```

Comment:
This fine as `Validate`: it is implied that it is validating the encoded portion.